src/components/schema-editor/schema-editor.component.tsx

```diff
@@ -2,49 +2,142 @@ import React from 'react';
 import AceEditor from 'react-ace';
 import 'ace-builds/webpack-resolver';
 import 'ace-builds/src-noconflict/ext-language_tools';
-import { useTranslation } from 'react-i18next';
+import Ajv from 'ajv';
+import capitalize from 'lodash-es/capitalize';
+import debounce from 'lodash-es/debounce';
+import { Toggletip, ToggletipContent, ToggletipButton, ToggletipActions, Link } from '@carbon/react';
+import { WarningFilled, CheckmarkFilled } from '@carbon/react/icons';
 import styles from './schema-editor.scss';
+import { useStandardFormSchema } from '../../hooks/useStandardFormSchema';
 
+interface markerProps {
+  startCol: number;
+  startRow: number;
+  endCol: number;
+  endRow: number;
+  className: string;
+  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents, @typescript-eslint/no-explicit-any
+  type: string | any;
+  text: string;
+}
 interface SchemaEditorProps {
   isLoading: boolean;
-  invalidJsonErrorMessage: string;
+  validation: boolean;
   onSchemaChange: (stringifiedSchema: string) => void;
   stringifiedSchema: string;
+  errors: Array<markerProps>;
+  setErrors: (errors: Array<markerProps>) => void;
 }
 
-const SchemaEditor: React.FC<SchemaEditorProps> = ({ invalidJsonErrorMessage, onSchemaChange, stringifiedSchema }) => {
-  const { t } = useTranslation();
+const SchemaEditor: React.FC<SchemaEditorProps> = ({
+  onSchemaChange,
+  stringifiedSchema,
+  errors,
+  setErrors,
+  validation,
+}) => {
+  const { schema } = useStandardFormSchema();
+
+  const validateJSON = (content: string, schema) => {
+    try {
+      const ajv = new Ajv({ allErrors: true, jsPropertySyntax: true });
+      const validate = ajv.compile(schema);
+      const parsedContent = JSON.parse(content);
+      const isValid = validate(parsedContent);
+
+      if (!isValid) {
+        const jsonLines = content.split('\n');
+        const errorMarkers = validate.errors.map((error) => {
+          let lineNumber = -1;
+          const instancePath = error.instancePath.replace(/^\./, '');
+          for (let i = 0; i < jsonLines.length; i++) {
+            if (jsonLines[i].includes(instancePath)) {
+              lineNumber = i;
+              break;
+            }
+          }
+          const message =
+            error.keyword === 'type'
+              ? `Invalid Type: ${error.instancePath.substring(1)} ${error.message}`
```

Comment:
Ditto

Reply:
@denniskigen do you mean translating this one too?